### ISSUE TITLE
Optimize shell acquisition order

### ISSUE THREAD

**Issue by shanjunmei** [NHAS/reverse_ssh#87]
https://github.com/NHAS/reverse_ssh/blob/581c6c5b559f71de1022a1d43c7d6e11a9fac7e3/internal/client/handlers/shell.go#L24

Here, if the shell list is read from the system configuration file(/etc/shells), when the client connection come in, take it in reverse order. If it is guessed, take the first one in order, and the experience will be better, because it seems that the list in front of the system configuration is generally It's all sh, a bit too ugly. . ?

**Comment by NHAS** [comment]
Hey, this is a good idea. 
However, I am a little worried that if I reverse `shells` we'll end up with a non-standard shell like git shell, (or screen).

It might be better to rethink the search entirely and use `shells` as an indicator of known good shells, rather than as a definitive list of shells.

Regardless. I agree that the default of sh is a pain

**Comment by NHAS** [comment]
This has now been fixed/updated, a list of good shells will be searched for by default. Failing to find any of those good shells in `PATH` they will be looked up in the `/etc/shells` if that files, then a bruteforce will be carried out in common executable containing directories. 

```golang
	goodShells = []string{
		"zsh",
		"bash",
		"fish",
		"dash",
		"ash",
		"csh",
		"ksh",
		"tcsh",
		"sh",
	}
```

List of common good shells that will be searched for (note the order, `sh` last).


This issue will be considered closed when you give confirmation that this fixes what you were experiencing, or when I merge this into master (as I've tested it myself)
